crates/cairo-lint-core/tests/test_files/collapsible_if/collapsible_if

```diff
@@ -0,0 +1,201 @@
+//! > collapsible if in boolean conditions
+
```

Comment:
what happens for that case:
```cairo
fn main() {
    let x = true;
    let y = true;
    let z = false;

    if x || z {
        if y && z {
            println!("Hello");
        } else {
            println!("World");
        }
    }
}
```
and in that case
```cairo
fn main() {
    let x = true;
    let y = true;
    let z = false;

    if x || z {
        if y && z {
            println!("Hello");
        }
        println!("World");
    }
}
```

Reply:
For the first case: The `collapsible_if` diagnostic should NOT be triggered, because combining the `if` conditions would break the logic of the `else` clause. I'm going to add a new test to ensures that the collapsible_if lint does not incorrectly suggest combining if statements when doing so would change the program’s behavior, especially when there's an else block involved.

Reply:
For the second case: If we combine the if as in (x || z) && (y && z), the logical flow would change because the statement println!("World"); would be executed only if x || z is true, but it would no longer be separated from the internal if, which is not desired.